src/components/Modals/SetAclPermissionsModal.jsx

```diff
@@ -125,28 +141,34 @@ const SetAclPermissionsModal = ({ showModal, setShowModal, dataset }) => {
             </Select>
           </FormControl>
           <br />
-          <FormControl fullWidth sx={{ marginBottom: '1rem' }}>
-            <TextField
-              id="set-acl-to"
-              name="setAclTo"
-              value={permissionState.podUrlToSetPermissionsTo}
-              onChange={(e) =>
-                setPermissionState({
-                  ...permissionState,
-                  podUrlToSetPermissionsTo: e.target.value
-                })
-              }
-              placeholder={permissionState.podUrlToSetPermissionsTo}
-              label="Enter podURL"
-              required
-              error={permissionState.podUrlToSetPermissionsTo === podUrl}
-              helperText={
-                permissionState.podUrlToSetPermissionsTo === podUrl
-                  ? 'Cannot share to your own pod.'.toUpperCase()
-                  : ''
-              }
-            />
-          </FormControl>
+          <Autocomplete
+            id="set-acl-to"
+            name="setAclTo"
+            data-testid="newShareWith"
+            sx={{ marginBottom: '1rem' }}
+            freeSolo
+            fullWidth
+            required
+            autoFocus
+            value={shareName?.person ?? permissionState.webIdToSetPermsTo}
+            disablePortal
+            autoSelect
+            options={contactListOptions}
+            onChange={(event, newValue) => {
+              setPermissionState({
+                ...permissionState,
+                webIdToSetPermsTo: newValue.id ?? newValue
+              });
+            }}
+            placeholder={permissionState.podUrlToSetPermissionsTo}
+            error={permissionState.podUrlToSetPermissionsTo === podUrl}
```

Comment:
- renderInput isn't defined until the component is rendered, so I hardcoded `placeholder={"WebID to share with"}`

Reply:
> The state `permissionState` in line 47 don't have the values `podUrlToSetPermissionsTo` set to them. I believe it's supposed to be `webIdToSetPermsTo` now that podUrl got updated to webIds.
> 
> As for the placeholder, I don't think we need that anymore since we could just use the text from renderInput, which is "WebID to share with" and the text itself is pretty helpful.

- renderInput isn't defined until the component is rendered, so I hardcoded `placeholder={"WebID to share with"}`